src/plugins/data/public/ui/dataset_navigator/dataset_navigator.tsx

```diff
@@ -55,49 +54,51 @@ export interface DataSetNavigatorProps {
   dataSet?: DataSetContract;
 }
 
-interface DataSetNavigatorState {
-  isMounted: boolean;
-  isOpen: boolean;
-  isLoading: boolean;
-  isExternalDataSourcesEnabled: boolean;
-  indexPatterns: any[];
-  dataSources: SimpleDataSource[];
-  externalDataSources: SimpleDataSource[];
-  currentDataSourceRef?: SimpleDataSource;
-  currentDataSet?: SimpleDataSet;
-  cachedDatabases: any[];
-  cachedTables: SimpleObject[];
-}
-
 interface SelectedDataSetState extends SimpleDataSet {
   database?: any | undefined;
 }
 
-export const DataSetNavigator = (props: DataSetNavigatorProps) => {
-  const { savedObjectsClient, http, dataSet: dataSetManager } = props;
+export const DataSetNavigator: React.FC<DataSetNavigatorProps> = ({
+  savedObjectsClient,
+  http,
+  dataSet: initialDataSet,
+}) => {
   const searchService = getSearchService();
   const queryService = getQueryService();
   const uiService = getUiService();
   const indexPatternsService = getIndexPatterns();
   const notifications = getNotifications();
 
-  const { dataSet } = useDataSetManager({ dataSetManager: dataSetManager! });
-
-  const [navigatorState, setNavigatorState] = useState<DataSetNavigatorState>({
-    isOpen: false,
-    isLoading: false,
-    isMounted: false,
-    isExternalDataSourcesEnabled: false,
-    dataSources: [],
-    externalDataSources: [],
-    currentDataSourceRef: undefined,
-    currentDataSet: undefined,
-    indexPatterns: [],
-    cachedDatabases: [],
-    cachedTables: [],
-  });
-
-  const [selectedDataSetState, setSelectedDataSetState] = useState<SelectedDataSetState>();
+  const { dataSet } = useDataSetManager({ dataSetManager: initialDataSet! });
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isExternalDataSourcesEnabled, setIsExternalDataSourcesEnabled] = useState(false);
+  const [dataSources, setDataSources] = useState<SimpleDataSource[]>([]);
+  const [externalDataSources, setExternalDataSources] = useState<SimpleDataSource[]>([]);
+  const [indexPatterns, setIndexPatterns] = useState<any[]>([]);
+  const [cachedDatabases, setCachedDatabases] = useState<any[]>([]);
+  const [cachedTables, setCachedTables] = useState<SimpleObject[]>([]);
+  const [selectedDataSetState, setSelectedDataSetState] = useState<
+    SelectedDataSetState | undefined
+  >(
+    dataSet
+      ? {
+          id: dataSet.id,
+          title: dataSet.title,
+          type: dataSet.type,
+          timeFieldName: dataSet.timeFieldName,
+          fields: dataSet.fields,
+          dataSourceRef: dataSet.dataSourceRef
+            ? {
+                id: dataSet.dataSourceRef.id,
+                name: dataSet.dataSourceRef.name,
+                type: dataSet.dataSourceRef.type,
+              }
+            : undefined,
+        }
+      : undefined
+  );
```

Comment:
Nit: 
```suggestion
dataSet && {
  ...dataSet,
  dataSourceRef: dataSet.dataSourceRef && {
    ...dataSet.dataSourceRef
  }
}
```

Reply:
i can address this originally. i didn't want to set the fields